### ISSUE TITLE
[Enhancement] Wildcard support for zos_data_set

### ISSUE THREAD

**Issue by Tom-Slattery** [ansible-collections/ibm_zos_core#798]
### Is there an existing issue for this?

- [X] There are no existing issues.

### Ansible module

zos_data_set

### Enhancement or feature description

Allowing wildcard support for zos_data_set would allow actions to affect multiple datasets without each one needing to be individually specified in the step. The syntax could be made to match the way datasets are interacted with on z/OS: 

% means any 1 character
* means any 1 qualifier (thus 1 through 8 characters, no period)
** means 0 or more qualifiers

(this does not correspond to how wildcards function on other platforms e.g. linux)

I think this functionality already exists in zos_find.

**Comment by ddimatos** [comment]
@Tom-Slattery - thank you for the suggestion. I am digging into this a bit more and looking at the various states supported, I am struggling to see a use case where a wild card would be needed for `state=present`, i can set it would make sense to support wildcards for states (catalog, uncataloged, absent) but for `present` it seems to be more inline with a 1:1 operation. 

Any thoughts on present? 

For wildcards doc for z/OS:
- https://www.ibm.com/docs/en/zos/2.1.0?topic=naming-wild-cards
- https://www.ibm.com/docs/en/zos/2.1.0?topic=naming-wild-card-restrictions

**Comment by Tom-Slattery** [comment]
Yes I think the wildcard use case only makes sense when the name field is being used to specify an existing dataset (if I've understood it right the name field's usage is subtly different depending on the state, in state=absent you're using it as a search query to select an existing DS, but for state=present you're using it to specify a new dataset name?)

I wonder if it would make sense to have a different field that supports wildcards for interacting with existing datasets? That field could then be inactive for state:present

Maybe something like:

- name: Create a dataset
  zos_data_set:
    name: ds.name.no.wildcard.allowed
    state: present

- name: Delete a dataset
  zos_data_set:
    query: I.Support.Wildcards.*
    state: absent

Otherwise could just have the name field be treated differently between state:absent and state:present - think either would work fine as a solution

**Comment by ddimatos** [comment]
@Tom-Slattery - you are correct when stating: " but for state=present you're using it to specify a new dataset name?)", both in agreement then. 

As for a new option `query`, ill share the idea with the team, but I think either way doc that explains the usage of wild cards supported by the various states or a new option will work. 

Keep an eye for when this will be planned, not likely a Q3 item but will see what we can do.
